reanalysis/hail_filter_and_label.py

```diff
@@ -690,23 +696,40 @@ def informed_repartition(
     https://discuss.hail.is/t/best-way-to-repartition-heavily-filtered-matrix-tables/2140
 
     :param matrix:
-    :param post_annotation:
     :param temporary_path:
+    :param post_annotation:
+    :param tiny:
     :return: repartitioned matrix
     """
 
+    tmp_2 = temporary_path + '_2'
+
     # calculate partitions, falling back to 1 partition if size is too small
     current_rows = matrix.count_rows()
     if post_annotation:
         partitions = current_rows // 200000 or 1
+    elif tiny:
+        partitions = 10
     else:
         partitions = current_rows // 100000 or 1
 
-    # repartition with the specified # partitions
-    matrix.repartition(n_partitions=partitions, shuffle=True)
+    matrix.checkpoint(tmp_2, overwrite=True)
+    hl.read_matrix_table(tmp_2, _n_partitions=partitions).write(
+        temporary_path, overwrite=True
+    )
+    matrix = hl.read_matrix_table(temporary_path)
```

Comment:
Might be worth checking whether a simple `repartition` call is really that much worse when using the service backend.

Reply:
I've re-written this to use a simpler repartition, at the moment with/without repartitioning the only failure appears at the `collect()` step, so it's hard to judge efficacy.